### ISSUE TITLE
CloudSave plugin doesn't build alongside the built-in "share" plugin [or different plugins] - android export debug apk

### ISSUE THREAD

**Issue by therealPaulPlay** [Chadori/construct-master-collection#615]
## Request Type
The cloudsave plugin does not build when you use construct's exporter and also have the Share addon in the project. My settings were: debug apk, android 9+, require vibrate and fullscreen permission. 

Edit: I found out that other plugins can cause the same issue, didn't test all now but with 30 plugins installed (that are not the share plugin) it also fails to build/export so the share plugin is not the only one. I can confirm that it exports without the cloud save plugin and these other 30 plugins.

## Describe the bug
The important part of the build error is: 
```
Failed to install 'cordova-plugin-cloud-settings': Error: There was a conflict trying to modify attributes with <edit-config> in plugin cordova-plugin-cloud-settings. The conflicting plugin, undefined, already modified the same attributes. The conflict must be resolved before cordova-plugin-cloud-settings can be added. You may use --force to add the plugin and overwrite the conflicting attributes.
at PlatformMunger.add_plugin_changes (~~/ConfigChanges.js:120:19)
```

and 

```
at process.processTicksAndRejections (node:~~/task_queues:77:11)
Failed to restore plugin "cordova-plugin-cloud-settings". You might need to try adding it again. Error: Error: There was a conflict trying to modify attributes with <edit-config> in plugin cordova-plugin-cloud-settings. The conflicting plugin, undefined, already modified the same attributes. The conflict must be resolved before cordova-plugin-cloud-settings can be added. You may use --force to add the plugin and overwrite the conflicting attributes.
```

## Reproducible .c3p or .capx
(Removed due to uploading bundled plugin)

## Steps to Reproduce
1. Create a new project with the cloudsave addon + share addon
2. export the project to android, use debug apk
3. Error

## Expected behavior
It should export just fine. 

## Construct Editor
 - Construct 3

## Construct Runtime
 - Construct 3

 ## Construct 3 Version 
r336

**Comment by Chadori** [comment]
Do not upload bundled plugins!

**Comment by Chadori** [comment]
I've fixed it by removing and adding back the **Share** object.

![image](https://user-images.githubusercontent.com/31282960/235372219-c279cd9b-0d6f-4c11-ad02-479237ae4529.png)

The issue here is due to how Construct 3 loads Cordova plugins, which is internal, and there's no way to control it during export through the Addon SDK. Hence, we may as well just call this a corrupted project, so the solution is to reorganize the plugins list by removing and then adding back either plugins.

New projects also work fine, as shown below.

![image](https://user-images.githubusercontent.com/31282960/235372357-633c1c98-7e0f-4839-b661-bfedcf6a3b24.png)

**Comment by Chadori** [comment]
Tldr; I'm afraid it's an engine export limitation, and an easy fix is removing and adding back the **Share** object. Below is the fixed version of your sent project (without bundled plugins).

[Fixed version.zip](https://github.com/Chadori/Construct-Master-Collection/files/11362158/Fixed.version.zip)

If you try in a new project, it should work fine as well.

**Comment by therealPaulPlay** [comment]
> Do not upload bundled plugins!

I am sorry, I forgot this is public. So you are telling me, this is „due to how construct loads plugins“ - what does that mean exactly? Is there an easy way to figure out which plugins I need to add again? Removing just the share plugin from my project did not do it for me.

I‘ve a lot of addons in my main project, over 10k actions etc. I can‘t just remove and re-add all addon related objects, that is unrealistic. 

I have never had this issue with any other plugin before, and I have installed and also made quite a few. Without your cloudsave plugin, it is all working fine. 

Is there a real solution to this problem? I need more info if it is an issue with my project.

**Comment by therealPaulPlay** [comment]
Maybe there is a way I can manually edit the project files to make it work? Please elaborate what is actually going on.

**Comment by Chadori** [comment]
> Maybe there is a way I can manually edit the project files to make it work? Please elaborate what is actually going on.

Sorry for the delay.

If the previous didn't work for you, I think there's another way that's easy as well. 
Could you upload that reproducible project file again, the broken one, although make sure to not enable bundles mode.

**Comment by therealPaulPlay** [comment]
Thanks for your response! 
I've done some research and it is actually a cordova bug with the <edit-config> tags. If a different plugin needs to change the tags first, this will happen (according to https://github.com/dpa99c/cordova-plugin-cloud-settings/issues/13)
I tested out every included plugin with construct and the ones that make changes using this tag are: Share, Google Play.
If someone else faces the same issue, they need to re-add these objects, which surely is annoying but unless cordova fixes this issue (from what I understand) we can't do anything about it apart from running cordova with --force locally.

**Comment by therealPaulPlay** [comment]
[BluelifeCloudsaveTest - Copy.zip](https://github.com/Chadori/Construct-Master-Collection/files/11414983/BluelifeCloudsaveTest.-.Copy.zip) Here is the broken project.

**Comment by Chadori** [comment]
After removing the **Share** object, saving the project, adding back the **Share** object and finally saving the project again; Github had shown that it put the **Cloud Save** object back on top of the list like how it is shown in the Construct 3 editor.

`project.c3proj` file:
![image](https://user-images.githubusercontent.com/31282960/236678463-02829fd3-8967-450f-9770-c6f67091d66b.png)

Construct 3 editor:
![image](https://user-images.githubusercontent.com/31282960/236678675-c23518da-8a3e-4c4d-8e37-5ad9ca9c35a1.png)

If a build error happens again, you can just edit the `project.c3proj` file and put the `"CloudSave"` object on top of the list (array) relative to the other mobile plugins that utilize Cordova.

As explained before, unfortunately this is a Construct 3 editor limitation that we have no choice but to overcome. The Construct Team is very small, and unfortunately requests for features to overcome issues like this cannot be prioritized by them. Therefore, if an issue like this happens again, try fixing it manually first as explained above.

If the issue still persists, feel free to comment below and I can try to provide other solutions.

**Comment by Chadori** [comment]
It successfully builds now:

![image](https://user-images.githubusercontent.com/31282960/236679098-1edb06fc-9465-4514-bae3-79289d69f10e.png)

**Comment by Chadori** [comment]
Just in case I wasn't making myself clear, you can edit the `project.c3proj` with any text editor.

![image](https://user-images.githubusercontent.com/31282960/236679395-b679e77e-63d7-463c-9497-8ad3425b7bb9.png)

Find the `CloudSave` object, or whatever name you've given the **Cloud Save** plugin in the Construct 3 editor.

![image](https://user-images.githubusercontent.com/31282960/236679427-970abd7b-3647-410e-af4e-83309db226d8.png)

Move it on top relative to other mobile plugins that utilize Cordova. Or, to be safe, move it to the top-most hierarchy.

![image](https://user-images.githubusercontent.com/31282960/236679481-3602c26a-f4b8-4293-8d47-7dd91178dbf2.png)

Now, reopen the Construct 3 project in the Construct 3 editor. It should now build successfully.

**Comment by therealPaulPlay** [comment]
Thanks for the detailed explanation, based on the first screenshot I was able to find the right part in Notepad++ eventually ;) I moved it to the top, opened the new project folder -> works! Thank you very much!!
